index.bs

```diff
@@ -12337,18 +12336,77 @@ The value of the [=function object=]’s <code class="idl">name</code> property
 
 <h3 id="es-platform-objects">Platform objects implementing interfaces</h3>
 
+<div algorithm>
+    An ECMAScript value |value| <dfn id="es-is-platform-object">is a platform object</dfn>
+    if [$Type$](|value|) is Object and if |value| has a \[[PrimaryInterface]] internal slot.
+</div>
+
+<div algorithm>
+    An ECMAScript value |value| <dfn export>implements</dfn> an [=interface=] |interface|
+    if |value| [=is a platform object=] and one of the following is true:
+    <ul>
+        <li>|value|.\[[PrimaryInterface]] is |interface|.</li>
+        <li>The set of [=inherited interfaces=] of |value|.\[[PrimaryInterface]] contains |interface|.</li>
+    </ul>
+</div>
+
+<div class="note">
+    Specifications use various wordings to reference the "[=implements=]" algorithm,
+    such as "|object| is an |interface| object", to mean, "|object| implements |interface|".
+</div>
+
 Every [=platform object=] is associated with a global environment, just
 as the [=initial objects=] are.
 It is the responsibility of specifications using Web IDL to state
 which global environment (or, by proxy, which global object) each platform
 object is associated with.
 
+<div algorithm>
+    To <dfn>create an object implementing the interface</dfn> |interface|, with optional
+    intenal slots |slots|, for an interface which is not declared with the
+    [{{Global}}] [=extended attribute=], perform the following steps:
+
+    1. If |slots| is provided, append \[[PrimaryInterface]] to |slots|.
+    1. Otherwise, let |slots| be &laquo; \[[PrimaryInterface]] &raquo;.
+    1. Let |proto| be the [=interface prototype object=] of |interface|.
```

Comment:
So this algorithm doesn't do subclassing.  I assume that's because it's meant to be done from specs that really want the default prototype?

Which global's interface prototype object is used here?  Seems like the right global should be input to this algorithm.

Reply:
What do you mean by "do subclassing"? Are there any specs that don't want the default prototype?

My guess is that the current global object would make sense, but there's a lot to test to see if this is solidly web reality...

Reply:
> What do you mean by "do subclassing"?

I mean `class Foo extends Event` in JS.

> Are there any specs that don't want the default prototype?

Well, `new Foo` in my example does not want the default prototype.  What prototype would it get and why, as things stand now?

Reply:
Oh, of course, I was thinking about interfaces extending other interfaces. I didn't realize that browsers actually used new.target from platform constructors this way; changed the draft spec to reflect that.